packages/block-editor/src/components/grid/grid-visualizer.js

```diff
@@ -62,6 +62,17 @@ const GridVisualizerGrid = forwardRef(
 				observer.observe( element );
 				observers.push( observer );
 			}
+
+			const mutationObserver = new window.MutationObserver( () => {
+				setGridInfo( getGridInfo( gridElement ) );
+			} );
+			mutationObserver.observe( gridElement, {
+				attributeFilter: [ 'style' ],
```

Comment:
I wonder if 'class' should also be added to this list. Change block style variations or adding a custom classname could also result in a change of top padding/border.